src/libsystemd/sd-network/network-util.c

```diff
@@ -13,20 +13,32 @@
 #include "strv.h"
 
 bool network_is_online(void) {
-        _cleanup_free_ char *carrier_state = NULL, *addr_state = NULL;
+        _cleanup_free_ char *online_state = NULL, *carrier_state = NULL, *addr_state = NULL;
```

Comment:
Please move the declarations of `carrier_state` and `addr_state` to the `else if (state < 0)` block below.